frontend/src/components/GameItem/GameItem.jsx

```diff
@@ -0,0 +1,49 @@
+import React from "react";
+import { Button } from "@mui/material";
+
+import { generateButtonProperties } from "../GameList/GameList.utils";
+import { joinGame, viewGame } from "../GameList/GameList.actions";
+import "./GameItem.styles.scss";
+
+import { isSpectator } from "../../pte-specifics/helpers/game.helpers";
+
+function GameItem({ game, type, walletResource, navigate }) {
+
+    if (!game) {
+        return null;
+    }
+
+    const buttonProperties = generateButtonProperties({
+        actions: {
+            join: () => joinGame({
+                gameAddress: game?.game_address,
```

Comment:
Probably dont need the `?` now because you check for truthy above right?

Reply:
Ahh yes of course - I've put extra guards in all over the place, so there'll be code like this that'll need to be trimmed in the next iteration / feature. Noted 💪